pytorch_lightning/callbacks/pruning.py

```diff
@@ -381,13 +385,18 @@ def on_train_epoch_end(self, trainer, pl_module, *args):
         ):
             self.apply_lottery_ticket_hypothesis()
 
-    def on_train_end(self, *args):
+    def on_train_end(self, trainer, pl_module: LightningModule):
         if self._make_pruning_permanent:
-            self.make_pruning_permanent()
+            rank_zero_debug("`on_train_end`. Pruning permanently...")
+            self.make_pruning_permanent(pl_module)
 
-    def on_save_checkpoint(self, *args):
+    def on_save_checkpoint(self, trainer, pl_module: LightningModule, checkpoint: Dict[str, Any]):
         if self._make_pruning_permanent:
-            self.make_pruning_permanent()
+            rank_zero_debug("`on_save_checkpoint`. Pruning permanently...")
+            # prune a copy so training can continue with the same buffers
+            copy = deepcopy(pl_module)
```

Comment:
Just to make sure, does this happen on the CPU mem?

Reply:
Changed!